### PR TITLE
Add support for field validation

### PR DESCRIPTION
**Pull request by DavisRayM**
# Changes

- Add `FIELD_VALIDATION_REGEX`
- Add ability to validate fields before creation of a user

### CODE REVIEW

File: tests/test_viewsets.py [hunk 1]
```diff
@@ -187,6 +188,22 @@ def test_validates_username(self):
...
             )
 
+            # Test error returned when username doesn't match regex
+            mock_func.return_value = {
+                "given_name": "jane",
+                "family_name": "doe",
+                "email": "jane@doe.com",
+                "preferred_username": "12345",
+            }
+            data = {"id_token": "ssad9012.fdfdfdswg4gdfs.sadadsods"}
```
Comment (WinnyTroy) [inline tests/test_viewsets.py:198]: The id_token seems to be used acrosee multiple tests, can we have this stored in a settings file?

**Review by WinnyTroy — CHANGES_REQUESTED** [review]

File: tests/test_viewsets.py (hunk 1 above)
Comment (DavisRayM) [reply tests/test_viewsets.py:198]: I don't think this should be in settings... It's just a random mocked value... that doesn't particular do much even if it's changed

File: README.md [hunk 2]
```diff
@@ -40,6 +40,7 @@ OPENID_CONNECT_VIEWSET_CONFIG = {
     "USE_AUTH_BACKEND": False,
     "AUTH_BACKEND": "",  # Defaults to django.contrib.auth.backends.ModelBackend
     "USE_RAPIDPRO_VIEWSET": False,
+    "FIELD_VALIDATION_REGEX": {"username": "(?!^\d+$)^.+$"} # This is the default validation
```
Comment (denniswambua) [inline README.md:43]: Can we have a comment describing the validation?

Comment (DavisRayM) [reply README.md]: Changed in the latest commit

Comment (denniswambua) [reply README.md]: @DavisRayM I suggest we have a descriptive message when the validation failed. Something like
```
`"FIELD_VALIDATION_REGEX": [{
          "Field": "username",
          "regex" : "(?!^\d+$)^.+$",
          "message": "Username should only contain alpha numeric characters"
}]
```
I feel this message `raise ValueError(f'Invalid `{k}` value `{data[k]}`')` is too ambiguous. We need to be raising more descriptive error messages.

**Review by denniswambua — APPROVED** [review]

**Pull request merged**